pi-top/Further-Link
# Support bluetooth connections

| Status  | Ticket/Issue |
| :---: | :--: |
| Hold | [Ticket](https://pi-top.atlassian.net/browse/SWE-186) |


#### Main changes

Support bluetooth connections by adding a GATT server with a service and several characteristics to interact with the frontend app.

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
- Download deb
- Make sure that the unstable apt repository is installed. You can install it by running:
```
sudo apt update
sudo apt install pi-top-os-unstable-apt-source
# Run again to update source
sudo apt update
```
- Install deb artifact by running `sudo apt install ./<file>`. It should automatically fetch the new dependencies from the unstable repository.
- Restart the further-link and bluetooth services
```
sudo systemctl daemon-reload
sudo systemctl restart further-link
sudo systemctl restart bluetooth
```
- Install miniscreen app using the deb file from [this PR](https://github.com/pi-top/pi-top-4-Miniscreen/pull/318).
- Enable pairing mode from Overview page of miniscreen app
- Use [this frontend branch](https://github.com/pi-top/further-frontend/pull/1847) for testing.

#### Tag anyone who definitely needs to review or help
-
